testing/oak_echo_bin/src/main.rs

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#![no_std]
+#![no_main]
+#![feature(alloc_error_handler)]
+#![feature(core_c_str)]
+#![feature(core_ffi_c)]
+
+extern crate alloc;
+
+use alloc::boxed::Box;
+#[cfg(all(feature = "raw_server", not(feature = "idl_server")))]
+use alloc::{vec, vec::Vec};
+use core::panic::PanicInfo;
+use log::info;
+use oak_baremetal_communication_channel::Channel;
+
+mod asm;
+mod bootparam;
+
+#[cfg(all(feature = "raw_server", not(feature = "idl_server")))]
+const MESSAGE_SIZE: usize = 1;
+
+#[no_mangle]
+pub extern "C" fn rust64_start(_rdi: u64, rsi: &bootparam::BootParams) -> ! {
+    let channel = oak_baremetal_kernel::start_kernel(rsi);
+    info!("In main!");
+    start_echo_server(channel)
+}
+
+fn start_echo_server(mut channel: Box<dyn Channel>) -> ! {
+    /// Starts an echo server that reads single bytes from the channel and writes
+    /// them back.
+    #[cfg(all(feature = "raw_server", not(feature = "idl_server")))]
```

Comment:
I think that one of these should not include the `not` part, otherwise this functions will be empty when clippy analyses it with all features enabled.

Reply:
Done